app/models/ems_event.rb

```diff
@@ -338,6 +338,29 @@ def ems_refresh_target
     ext_management_system
   end
 
+  def src_container_replicator
```

Comment:
`EmsEvent` is better to be the top level generic class, not containing type specific logic. Is there a way to separate the container specific code to subclass?